src/script/interpreter.cpp

```diff
@@ -606,8 +606,17 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     // To provide for future soft-fork extensibility, if the
                     // operand has the disabled lock-time flag set,
                     // CHECKSEQUENCEVERIFY behaves as a NOP.
-                    if ((nSequence & CTxIn::SEQUENCE_LOCKTIME_DISABLE_FLAG) != 0)
+                    // If uninterpreted fields are set, also treat as a NOP.
+                    // This also covers CTxIn::SEQUENCE_FINAL since all bits are
+                    // set there.
+                    const CScriptNum extra_fields = nSequence &
+                        ~(CTxIn::SEQUENCE_LOCKTIME_TYPE_FLAG | CTxIn::SEQUENCE_LOCKTIME_MASK);
+                    if ((nSequence & CTxIn::SEQUENCE_LOCKTIME_DISABLE_FLAG) != 0 || extra_fields != 0) {
```

Comment:
adding this `extra_fields` check should result in a fork now, no?

Reply:
You're totally right, my bad.

We need to handle it separately and still pass it to CheckSequence even though the fields are ignored.

Reply:
Ok I think the patch i added properly only checks the extra flags during standardness interpreter rules.

Reply:
I was wrong on that; the prior policy.cpp i had was not exempting final sequence. new version should be (hopefully) correct.